scripts/test-remote-dev-codex-runtime.py

```
@@ -272,6 +309,261 @@ def test_successful_publication_retains_previous_generation(self):
+            ):
+                self.m.update_runtime(yes=True)
+            download.assert_not_called()
+            self.assertEqual(list(fixed.iterdir()), [])
+
+    def test_candidate_rejects_a_nontraversable_parent(self):
+        with tempfile.TemporaryDirectory() as text:
+            fixed = self.staging_root(Path(text))
+            with mock.patch.object(self.m, "STAGING_ROOT", fixed), mock.patch.object(
+                self.m, "probe_staging_execution"
+            ):
+                with self.m.update_staging() as staging:
```

Ruff (0.16.1)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)